src/examples/mfem_common_plane.cpp

```diff
@@ -0,0 +1,429 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+#include <set>
+
+#ifdef TRIBOL_USE_UMPIRE
+// Umpire includes
+#include "umpire/ResourceManager.hpp"
+#endif
+
+// MFEM includes
+#include "mfem.hpp"
+
+// Axom includes
+#include "axom/CLI11.hpp"
+#include "axom/slic.hpp"
+
+// Redecomp includes
+#include "redecomp/redecomp.hpp"
+
+// Tribol includes
+#include "tribol/common/Parameters.hpp"
+#include "tribol/config.hpp"
+#include "tribol/interface/tribol.hpp"
+
+/// Simple central difference method.
+class CentralDiffSolver : public mfem::SecondOrderODESolver
+{
+public:
+  CentralDiffSolver(const mfem::Array<int>& bc_vdofs_)
+  : bc_vdofs { bc_vdofs_ },
+    first_step { true } {}
+
+  void Step(mfem::Vector& x, mfem::Vector& dxdt, double& t, double& dt) override
+  {
+    // acceleration at t
+    f->SetTime(t);
+    if (first_step)
+    {
+      accel.SetSize(x.Size());
+      f->Mult(x, dxdt, accel);
+      first_step = false;
+    }
+
+    // velocity at t + dt/2
+    dxdt.Add(0.5*dt, accel);
+
+    // set homogeneous velocity BC at t + dt/2
+    SetHomogeneousBC(dxdt);
+
+    // set displacement at t + dt
+    x.Add(dt, dxdt);
+
+    // acceleration at t + dt
+    f->SetTime(t + dt);
+    f->Mult(x, dxdt, accel);
+
+    // velocity at t + dt
+    dxdt.Add(0.5*dt, accel);
+  }
+private:
+  mfem::Vector accel;
+  mfem::Array<int> bc_vdofs;
+  bool first_step;
+
+  void SetHomogeneousBC(mfem::Vector& dxdt) const
+  {
+    for (auto bc_vdof : bc_vdofs)
+    {
+      dxdt[bc_vdof] = 0.0;
+    }
+  }
+};
+
+/// Explicit solid mechanics update (lumped mass, optional damping)
+class ExplicitMechanics : public mfem::SecondOrderTimeDependentOperator
+{
+public:
+  ExplicitMechanics(
+    mfem::ParFiniteElementSpace& fespace, 
+    mfem::Coefficient& rho,
+    mfem::Coefficient& lambda,
+    mfem::Coefficient& mu,
+    std::unique_ptr<mfem::ParBilinearForm> damping_ = nullptr
```

Comment:
Comment on type of damping or how this is used would be helpful.

Reply:
Damping has been removed